Src/ILGPU/IR/Analyses/LoopInfo.cs

```diff
@@ -344,6 +493,22 @@ internal InductionVariableOperation(int index, Value value, T kind)
     /// </summary>
     public readonly struct InductionVariableBounds
     {
+        #region Static
+
+        /// <summary>
+        /// Tries to map a loop variable to an integer constant.
+        /// </summary>
+        /// <param name="value">The value to map to an integer bound.</param>
+        /// <returns>The mapped integer (if any).</returns>
+        private static int? TryGetIntegerBound(Value value) =>
+            value is PrimitiveValue primitive && value.BasicValueType.IsInt()
+            ? primitive.Int32Value
+            : (int?)null;
```

Comment:
SUGGESTION: `default` is the equivalent of `(int?)null` in this instance.
```suggestion
            : default;
```

Reply:
Hmm I tried this myself and `default` seems to be converted to an instance of `int?` with value `0` instead of having no value at all.

Reply:
Might be because the first part of the conditional is returning `int`. Try `default(int?)`.